astropy/io/ascii/core.py

```diff
@@ -395,16 +401,16 @@ class BaseSplitter:
 
     """
 
-    delimiter = None
+    delimiter: ClassVar[str | None] = None
```

Comment:
This gets set on instances, so I think `ClassVar` is not appropriate.

Reply:
Can you provide an example of that? I grepped for `self.delimiter` and there are indeed a few matches, but I didn't spot any uses of `delimiter` as an instance attribute in `BaseSplitter` subclasses, so they are not relevant for this annotation.

Reply:
Look for `.splitter.delimiter = `.  This shows lines like:
```
    if "delimiter" in kwargs:
        if kwargs["delimiter"] in ("\n", "\r", "\r\n"):
            reader.header.splitter = BaseSplitter()
            reader.data.splitter = BaseSplitter()
        reader.header.splitter.delimiter = kwargs["delimiter"]
        reader.data.splitter.delimiter = kwargs["delimiter"]
```